src/openeo_gfmap/fetching/generic.py

```diff
@@ -28,18 +28,36 @@
     "vapour-pressure": "AGERA5-VAPOUR",
     "wind-speed": "AGERA5-WIND",
 }
+AGERA5_STAC_MAPPING = {
+    "dewpoint_temperature_mean": "AGERA5-DEWTEMP",
+    "total_precipitation": "AGERA5-PRECIP",
+    "solar_radiation_flux": "AGERA5-SOLRAD",
+    "2m_temperature_max": "AGERA5-TMAX",
+    "2m_temperature_mean": "AGERA5-TMEAN",
+    "2m_temperature_min": "AGERA5-TMIN",
+    "vapour_pressure": "AGERA5-VAPOUR",
+    "wind_speed": "AGERA5-WIND",
+}
 KNOWN_UNTEMPORAL_COLLECTIONS = ["COPERNICUS_30"]
 
+AGERA5_TERRASCOPE_STAC = "https://stac.openeo.vito.be/collections/agera5_daily"
+
 
 def _get_generic_fetcher(
-    collection_name: str, fetch_type: FetchType, backend: Backend
+    collection_name: str, fetch_type: FetchType, backend: Backend, is_stac: bool
 ) -> Callable:
     band_mapping: Optional[dict] = None
 
     if collection_name == "COPERNICUS_30":
         band_mapping = BASE_DEM_MAPPING
     elif collection_name == "AGERA5":
         band_mapping = BASE_WEATHER_MAPPING
+    elif is_stac and (AGERA5_TERRASCOPE_STAC in collection_name):
+        if backend not in [Backend.TERRASCOPE, Backend.FED]:
+            raise ValueError(
```

Comment:
Should this error not be thrown if `is_stac==False` and the backend is not Terrascope or Fed?

Reply:
Yes you are right, I removed the check. Actually:

> From a discussion with @HansVRP we decided that GFMAP shouldn't take too much care of checking if a collection is valid or not in a certain backend but rather instead let the load_collection method raise the exception itself.